src/metabase/api/field.clj

```diff
@@ -71,24 +72,70 @@
   [id]
   (let [field (api/read-check Field id)]
     (if-not (field-should-have-field-values? field)
-      {:values {} :human_readable_values {}}
-      (create-field-values-if-needed! field))))
+      {:values []}
+      {:values (create-field-values-if-needed! field)})))
 
+(api/defendpoint POST "/:id/dimension"
+  "If `Field`'s special type derives from `type/Category`, or its base type is `type/Boolean`, return
+   all distinct values of the field, and a map of human-readable values defined by the user."
+  [id :as {{dimension-type :type dimension-name :name human_readable_field_id :human_readable_field_id} :body}]
+  {dimension-type         (s/enum "internal" "external")
+   dimension-name         su/NonBlankString
+   human_readable_field_id (s/maybe s/Int)}
```

Comment:
You can also use `su/IntGreaterThanZero` here because a negative or zero ID wouldn't make sense. Not a big difference but heads up that it exists